### PR TITLE
Update release process and `master` becomes the stable branch

### PR DESCRIPTION
**Pull request by shilman**
Issue: N/A

## What I did

Update `master` branch per the [release process v4 proposal](https://gist.github.com/shilman/0b5a4011f8a105a36b657a79736733df)

## How to test

Is this testable with Jest or Chromatic screenshots?
Does this need a new example in the kitchen sink apps?
Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

For maintainers only: Please tag your pull request with at least one of the following:
`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,8 +26,8 @@ It allows you to browse a component library, view the different states of each c
 
 README for:
 
-- [![Alpha](https://img.shields.io/npm/v/@storybook/core/alpha.svg)](https://github.com/storybooks/storybook)
-- [![Latest](https://img.shields.io/npm/v/@storybook/core/latest.svg)](https://github.com/storybooks/storybook/tree/release/4.0)
+- [![Alpha](https://img.shields.io/npm/v/@storybook/core/alpha.svg)](https://github.com/storybooks/storybook/tree/next)
```
Comment (Hypnosphi) [inline README.md]: this link can be leaved as is

**Comment by Hypnosphi** [comment]
Let's also update https://github.com/storybooks/storybook/blob/master/.github/PULL_REQUEST_TEMPLATE.md with a note that PR base shouldn't be changed from `next` to `base` unless it's absolutely necessary and that bugfix PRs will be cherry-picked to current release

**Comment by shilman** [comment]
@Hypnosphi updated

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -4,11 +4,13 @@ Issue:
...
-Does this need a new example in the kitchen sink apps?
-Does this need an update to the documentation?
+- Is this testable with Jest or Chromatic screenshots?
+- Does this need a new example in the kitchen sink apps?
+- Does this need an update to the documentation?
 
 If your answer is yes to any of these, please make sure to include it in your PR.
 
-For maintainers only: Please tag your pull request with at least one of the following:
+**For everybody:** Please submit all PRs to the `next` branch unless they are specific to current release. Storybook maintainers cherry-pick bug and documentation fixes into the `master` branch as part of the release process, so you shouldn't need to worry about this.
```
Comment (Hypnosphi) [inline .github/PULL_REQUEST_TEMPLATE.md]: let's include both notes in `<!--`/`-->` to hide them from actual PRs

Comment (Hypnosphi) [reply .github/PULL_REQUEST_TEMPLATE.md]: See e.g https://raw.githubusercontent.com/facebook/create-react-app/master/.github/ISSUE_TEMPLATE.md

**Review by Hypnosphi — APPROVED** [review]

**Pull request merged**